### PR TITLE
Endpoint to create a new post ( also completes user posts endpoint)

### LINKED ISSUE #4: Endpoint (GET) for user posts index

### LINKED ISSUE #5: Endpoint (POST) for creating a new post

### PR DESCRIPTION
**Pull request by mkrumholz**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Linter has passed locally and any fixes were made for failures
- [x] Travis build is passing
- [x] All tests passing locally

## Pull request type
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: 4,5

No post mutations, no ability to filter posts. This adds some of that functionality and explores the basics for how to flesh out CRUD actions for a given resource.

## What is the new behavior?

- Creates an endpoint to create a new post
- Completes the endpoint to get a specific user's posts

closes #4 
closes #5

### CODE REVIEW

**Review by JoePeterson51 — APPROVED** [review]

File: app/graphql/mutations/posts/create_post.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Mutations
+  module Posts
+    class CreatePost < ::Mutations::BaseMutation
+      argument :title, String, required: true
+      argument :description, String, required: true
+      argument :image, String, required: false
+      argument :user_id, Integer, required: true
+
```
Comment (clairefields15) [inline app/graphql/mutations/posts/create_post.rb:8]: @k-atwhite make sure we have some error handling for this on the FE and don't allow submission without title and description. I created an issue for it.

File: spec/graphql/mutations/posts/create_post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      expect(data[:updatedAt]).to be_present
+      expect(data[:user]).to eq({ id: user2.id.to_s, username: user2.username, avatar: user2.avatar })
+    end
+
+    def query(user_id:)
+      <<~GQL
+        mutation createPost {
+          createPost(input: {
+            title: "Help! A swarm!"
+            description: "Holy swarmoly, I could use a hand here!"
```
Comment (clairefields15) [inline spec/graphql/mutations/posts/create_post_spec.rb:44]: 😂

File: spec/graphql/mutations/posts/create_post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            image: "https://loremflickr.com/119/211/swarm"
+            userId: #{user_id}
+            }
+          ){
```
Comment (clairefields15) [inline spec/graphql/mutations/posts/create_post_spec.rb:48]: I'm a little confused about what is happening here, the difference between the input and the rest of the query, but I haven't looked into mutations yet!

**Review by clairefields15 — APPROVED** [review]
Looks good, a few questions but just for understanding

File: spec/graphql/mutations/posts/create_post_spec.rb (hunk 3 above)
Comment (mkrumholz) [reply spec/graphql/mutations/posts/create_post_spec.rb:48]: The input is what fields we need you to send us -- the rest is actually defining which fields you want returned in the response @clairefields15

**Pull request merged**